ros_buildfarm/sourcedeb_job.py

```diff
@@ -59,6 +60,25 @@ def get_sources(
              'the expected package version from the distribution file (%s)') %
             (source_version, pkg_version))
 
+    # If a tarball already exists reuse it
+    origtgz_version = pkg_version.split('-')[0]
+    debian_package_name = get_debian_package_name(rosdistro_name, pkg_name)
+    filename = '%s_%s.orig.tar.gz' % (debian_package_name, origtgz_version)
+
+    for repo in debian_repository_urls:
+        URL_TEMPLATE = '%s/pool/main/%s/%s/%s'
+        prefix = debian_package_name[0]
+        url = URL_TEMPLATE % (repo, prefix, debian_package_name, filename)
+
+        try:
+            output_file = os.path.join(sources_dir, '..', filename)
+            urlretrieve(url, output_file)
+            print("Found matching original tarball, "
+                  "downloading %s to %s" % (url, output_file))
+            continue
+        except:
+            print("No tarball found at %s, that's ok it will be rebuilt" % url)
```

Comment:
Please put single quotes around the placeholders.

The part of the message ("it will be rebuilt") might be misleading since it might still be downloaded from another repo. Therefore I would suggest to leave that part out.